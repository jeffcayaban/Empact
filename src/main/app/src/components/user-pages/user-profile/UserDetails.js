import React from 'react';
import { Col, Grid } from "react-bootstrap";
import Circle from "./Circle";

/**
 * Displays the details of a given user.
 */

class UserDetails extends React.PureComponent {

    render() {
        const { firstName, lastName, userName } = this.props;

        const userInitial = String(firstName).charAt(0);
        const fullName = `${firstName} ${lastName}`;
        const usernameToShow = `@${userName}`;

        return (
            <Grid>
                <Col xs={12} md={12}>
                    <div className={"centerStyle"}>
                        <Circle text={userInitial} />
                        <div>
                            <h1 align={"center"}>{fullName}</h1>
                            <h4 align={"center"} style={{ color: '#7c7c7c' }}>{usernameToShow}</h4>
                        </div>
                    </div>
                </Col>
            </Grid>
        );
    }
}

export default UserDetails;
