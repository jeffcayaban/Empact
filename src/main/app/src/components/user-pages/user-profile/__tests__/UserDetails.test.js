import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import UserDetails from "../UserDetails";

describe('UserDetails', () => {

    it('should render correctly', () => {
        const component = shallow(<UserDetails
            firstName={"testFirstName"}
            lastName={"testLastName"}
            userName={"testUserName"}
        />);
        expect(toJson(component)).toMatchSnapshot();
    });
});
