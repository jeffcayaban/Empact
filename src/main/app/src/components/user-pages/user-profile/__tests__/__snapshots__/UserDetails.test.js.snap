// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetails should render correctly 1`] = `
<Grid
  bsClass="container"
  componentClass="div"
  fluid={false}
>
  <Col
    bsClass="col"
    componentClass="div"
    md={12}
    xs={12}
  >
    <div
      className="centerStyle"
    >
      <Circle
        text="t"
      />
      <div>
        <h1
          align="center"
        >
          testFirstName testLastName
        </h1>
        <h4
          align="center"
          style={
            Object {
              "color": "#7c7c7c",
            }
          }
        >
          @testUserName
        </h4>
      </div>
    </div>
  </Col>
</Grid>
`;
